src/neural/cuda/network_cuda.cc

```diff
@@ -48,6 +48,35 @@ using namespace cudnn_backend;
 template <typename DataType>
 class CudaNetwork;
 
+static size_t getMaxAttentionHeadSize(const LegacyWeights& weights, int N) {
```

Comment:
don't need to solve right now - but I could imagine that the 'cudnn' implementation of attention head reuses the attention head block as it stands for cuda - and so this code may need to be moved to a shared location.

Reply:
yes, I think it can same code can be supported in cudnn backend with minimal modifications. I will handle it in separate PR. (are people still using cudnn backend over cuda?)

Reply:
there are old gpu users where cuda is less of an option due to memory usage of the winograd transformations ... and maybe some 1650 type gpus where cudnn is faster than cuda? I don't keep track of all the details...

Reply:
The 1650 may be fine after the workaround in #1675 (approved but not yet merged to avoid conflicts with this one), but all other GTX cards seem to benefit from cudnn.